frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -1156,24 +1141,14 @@ function ReaderBookmark:setHighlightedText(item, text)
             text = self.ui.document:getTextFromPositions(item.pos0, item.pos1).text
         end
     end
-    -- highlight
-    local hl = self.ui.highlight:getHighlightByDatetime(item.datetime)
-    hl.text = text
-    hl.edited = edited
-    -- bookmark
-    local index = self:getBookmarkIndexBinarySearch(item) or self:getBookmarkIndexFullScan(item)
-    local bm = self.bookmarks[index]
-    local is_auto_text_before = self:isBookmarkAutoText(bm)
-    bm.notes = text
-    if is_auto_text_before then
-        bm.text = self:getBookmarkAutoText(bm)
-    end
-    bm.edited = edited
+    -- annotation
+    local index = (self.filtered_mode or self.show_edited_only) and self.ui.annotation:getItemIndex(item) or item.idx
+    self.ui.annotation.annotations[index].text = text
+    self.ui.annotation.annotations[index].text_edited = edited
```

Comment:
I guess there, `item` is a Menu item, not itself a annotations item, right?
(Otherwise, you would not need to peek into `self.ui.annotation.annotations[index]` to update the same thing.)

Reply:
Yes

Reply:
If `filtered_mode` is off (we show all bookmarks), we can use `idx` generated by the Menu widget.
If not, we must get index via the binary search.